src/Control/Email/MailerSubscriber.php

```diff
@@ -0,0 +1,116 @@
+ *
+ * See https://symfony.com/doc/current/mailer.html#mailer-events for further info
+ */
+class MailerSubscriber implements EventSubscriberInterface
+{
+    use Injectable;
+    use Extensible;
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            SendMessageToTransportsEvent::class => 'onSendMessageToTransports',
+        ];
+    }
+
+    public function onSendMessageToTransports(SendMessageToTransportsEvent $event): void
+    {
+        $envelope = $event->getEnvelope();
+        /** @var SendEmailMessage $sendEmailMessage */
+        $sendEmailMessage = $envelope->getMessage();
+        $email = $sendEmailMessage->getMessage();
+        if (!($email instanceof Email)) {
+            throw new InvalidArgumentException('Message is not a ' . Email::class);
+        }
+        $this->applyConfig($email);
+        $this->updateUrls($email);
+        $this->extend('updateOnMessage', $email, $event);
+    }
+
+    private function applyConfig(Email $email): void
+    {
+        $sendAllTo = Email::getSendAllEmailsTo();
+        if (!empty($sendAllTo)) {
+            $this->setTo($email, $sendAllTo);
+        }
+
+        $ccAllTo = Email::getCCAllEmailsTo();
+        if (!empty($ccAllTo)) {
+            $email->addCc(...$ccAllTo);
+        }
+
+        $bccAllTo = Email::getBCCAllEmailsTo();
+        if (!empty($bccAllTo)) {
+            $email->addBcc(...$bccAllTo);
+        }
+
+        $sendAllFrom = Email::getSendAllEmailsFrom();
+        if (!empty($sendAllFrom)) {
+            $this->setFrom($email, $sendAllFrom);
+        }
+    }
+
+    private function setTo(Email $email, array $sendAllTo): void
+    {
+        $headers = $email->getHeaders();
+        $xTo = $this->convertAddressesToString($email->getTo());
+        $xCc = $this->convertAddressesToString($email->getCc());
+        $xBcc = $this->convertAddressesToString($email->getBcc());
+
+        // set default recipient and remove all other recipients
+        $email->to(...$sendAllTo);
+        $email->cc(...[]);
+        $email->bcc(...[]);
+
+        // store the old data as X-Original-* Headers for debugging
+        if ($xTo !== '') {
+            $headers->addMailboxHeader('X-Original-To', $xTo);
+        }
+        if ($xCc !== '') {
+            $headers->addMailboxHeader('X-Original-Cc', $xCc);
+        }
+        if ($xBcc !== '') {
+            $headers->addMailboxHeader('X-Original-Bcc', $xBcc);
+        }
+    }
+
+    private function setFrom(Email $email, array $sendAllFrom): void
+    {
+        $headers = $email->getHeaders();
+        $xFrom = $this->convertAddressesToString($email->getFrom());
+        if ($xFrom !== '') {
+            $headers->addMailboxHeader('X-Original-From', $xFrom);
+        }
+        $email->from(...$sendAllFrom);
+    }
+
+    private function convertAddressesToString(array $addresses): string
+    {
+        return implode(',', array_map(fn(Address $address) => $address->getAddress(), $addresses));
```

Comment:
Does this account for the name, if one was set?

Reply:
I've removed this code and refactored to use `addMailboxListHeader()` which uses `Address[]` as a param, this gives support to uses names in Address's

To illustrate, when I send an email with

```php
$email->cc(new Address('my-cc1@example.com', 'My CC1 name'), 'my-cc2@example.com');
```
In mailhog, the `cc` header by default is
`My CC1 name <my-cc1@example.com>, my-cc2@example.com`

When I set my .env file to 
```
SS_SEND_ALL_EMAILS_TO="env.to@example.com"
```

And I send the same email, now the `X-Original-Cc` header is also
`My CC1 name <my-cc1@example.com>, my-cc2@example.com`